File: android/app/src/main/java/com/lightningkite/khrysalistemplate/layouts/FloatingHexagonsDemoXml.kt
//
// FloatingHexagonsDemoXml.swift
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysalistemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalistemplate.R

class FloatingHexagonsDemoXml {

    lateinit var customView: com.lightningkite.khrysalis.views.CustomView
    lateinit var customViewDelegate: com.lightningkite.khrysalistemplate.vg.FloatingHexagonsDelegate
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.floating_hexagons_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        customView = view.findViewById<com.lightningkite.khrysalis.views.CustomView>(R.id.customView)
        customViewDelegate = view.findViewById<CustomView>(R.id.customView).delegate as com.lightningkite.khrysalistemplate.vg.FloatingHexagonsDelegate
        
        
        return view
    }
}

File: web/src/layout/FloatingHexagonsDemoXml.ts
//
// FloatingHexagonsDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'khrysalis/dist/views/html'
import { customViewSetDelegate } from 'khrysalis/dist/views/CustomView.actual'
import { FloatingHexagonsDelegate } from '../vg/FloatingHexagonsDemoVG.shared'
import htmlForDefault from './floating_hexagons_demo.html'
//! Declares com.lightningkite.khrysalistemplate.layouts.FloatingHexagonsDemoXml
export class FloatingHexagonsDemoXml {
    xmlRoot!: HTMLElement;
    customView!: HTMLCanvasElement;
    customViewDelegate!: FloatingHexagonsDelegate;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.customView = getViewById<HTMLCanvasElement>(view, "customView");
        if(this.customView){ this.customViewDelegate = new FloatingHexagonsDelegate(); customViewSetDelegate(this.customView, this.customViewDelegate); }
        return view
    }
}
